src/main/resources/templates/login.html

```diff
@@ -12,9 +12,10 @@ <h2>Login</h2>
         <p>Invalid username or password.</p>
     </div>
     <form action="/login" method="post">
```

Comment:
In oauth2Login(), you should check that the /login page correctly handles authorization and redirection errors after a successful login.